app.go

```diff
@@ -186,11 +186,55 @@ func (a *App) useShortOptionHandling() bool {
 	return a.UseShortOptionHandling
 }
 
+func checkDuplicateSubcommandNames(subcommands []Command, parentCommandName string) error {
+	subcommandNamesWithoutDuplicates := make(map[string]bool)
+	for _, s := range subcommands {
+		_, sExists := subcommandNamesWithoutDuplicates[s.Name]
+		if sExists {
+			// @todo: update the following error message to take into account that the parent command may be a subcommand
+			return fmt.Errorf("Your command %q contains multiple subcommands with the Name %q. Having multiple subcommands with the same name results in ambiguous behavior, so please make sure each subcommand in your command has a unique name.", parentCommandName, s.Name)
+		}
+		subcommandNamesWithoutDuplicates[s.Name] = true
+	}
+
+	for _, s := range subcommands {
+		if len(s.Subcommands) > 0 {
+			err := checkDuplicateSubcommandNames(s.Subcommands, s.Name)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (a *App) checkDuplicateCommandNames() error {
+	commandNamesWithoutDuplicates := make(map[string]bool)
+	for _, c := range a.Commands {
```

Comment:
⭐⭐ _(seriously consider)_ consider passing in `App.Commands` rather than making `checkDuplicateCommandNames` a function on `App`.